ambry-api/src/main/java/com.github.ambry/account/AccountService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.io.Closeable;
+import java.util.Collection;
+
+
+/**
+ * An {@code AccountService} is a component that can respond to queries for {@link Account} by id or name, and
+ * add/update {@link Account} for future queries. Deleting an {@link Account} is not supported currently. The
+ * {@link Account}s under this {@code AccountService} cannot have duplicate ids or names, and MUSt have their
+ * ids and names one-to-one mapped.
+ */
+public interface AccountService extends Closeable {
+
+  /**
+   * Gets an {@link Account} by its id.
+   * @param accountId The id of an {@link Account} to get.
+   * @return The {@link Account} with the specified id. {@code null} if such an {@link Account} does not exist.
+   */
+  public Account getAccountById(short accountId);
+
+  /**
+   * Gets an {@link Account} by its name.
+   * @param accountName The name of an {@link Account} to get.
+   * @return The {@link Account} with the specified name. {@code null} if such an {@link Account} does not exist.
+   */
+  public Account getAccountByName(String accountName);
+
+  /**
+   * Makes update for a collection of {@link Account}s. This will need to write through the updates all the way to
+   * the backend storage where this {@code AccountService} relies on. There could be cases that the {@link Account}
+   * to update conflicts with the existing {@link Account}(s). It is up to the implementation to take different
+   * policies to resolve conflict.
+   * @param accounts The collection of {@link Account}s to update. Cannot be {@code null}.
+   * @return {@code true} if the update operation is successful, {@code false} otherwise.
+   */
+  public boolean updateAccounts(Collection<Account> accounts);
+
+  /**
+   * Gets all the {@link Account}s in this {@code AccountService}. The {@link Account}s <em>MUST</em> have their
+   * ids and names one-to-one mapped.
```

Comment:
if you are looking for mapping, shouldn't the return type be a `Map` instead of `Collections`?